packages/core/src/components/NumberInput.tsx

```diff
@@ -14,49 +14,75 @@ const NumberInput: FC<Props> = ({
   defaultValue,
   ...props
 }) => {
-  const formatValueToStringNumber = useCallback(
-    (valueToFormat?: number | string, currentStringNumberValue?: string) => {
-      if (valueToFormat != null) {
-        if (isString(valueToFormat) && valueToFormat !== "") {
-          if (/^0[1-9]$/.test(valueToFormat)) {
-            return valueToFormat.slice(1);
-          } else if (/^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/.test(valueToFormat)) {
-            return valueToFormat;
-          } else if (currentStringNumberValue) {
-            return currentStringNumberValue;
-          }
-        } else if (isNumber(valueToFormat) && !isNaN(valueToFormat)) {
-          return valueToFormat.toString();
+  const formatValueToStringNumber = (
+    valueToFormat?: number | string,
+    currentStringNumberValue?: string
+  ) => {
+    if (valueToFormat != null) {
+      if (isString(valueToFormat) && valueToFormat !== "") {
+        if (/^0[1-9]$/.test(valueToFormat)) {
+          return valueToFormat.slice(1);
+        } else if (/^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/.test(valueToFormat)) {
+          return valueToFormat;
+        } else if (currentStringNumberValue) {
+          return currentStringNumberValue;
         }
+      } else if (isNumber(valueToFormat) && !isNaN(valueToFormat)) {
+        return valueToFormat.toString();
       }
+    }
 
-      return "0";
-    },
-    []
-  );
+    return "0";
+  };
 
-  const [stringNumberValue, setStringNumberValue] = useState(
+  const [currentStringNumberValue, setCurrentStringNumberValue] = useState(
     formatValueToStringNumber(value)
   );
 
   const handleChangeText = (newValue: string) => {
     const newStringNumberValue = formatValueToStringNumber(
       newValue,
-      stringNumberValue
+      currentStringNumberValue
     );
     const number = parseFloat(newStringNumberValue);
 
-    setStringNumberValue(newStringNumberValue);
+    setCurrentStringNumberValue(newStringNumberValue);
     onChangeText?.(number);
   };
 
+  /* set currentStringNumberValue directly to defaultValue if it exists on load
+  ( no need to use TextInput's defaultValue because its value is always controlled & set ) */
+  useEffect(() => {
+    const defaultStringNumberValue = formatValueToStringNumber(
+      defaultValue,
+      currentStringNumberValue
+    );
+
+    if (currentStringNumberValue !== defaultStringNumberValue) {
+      setCurrentStringNumberValue(defaultStringNumberValue);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why don't we want the dependencies here? Seems like whenever they change we'd want to run this?

Reply:
@adnelson you only set the defaultValue when you initialize the component.  It never goes back to this value.

Imagine if you set a default to 6 and the deleted the number, you would not want it set back to 6.

Reply:
@adnelson BTW

`// eslint-disable-next-line react-hooks/exhaustive-deps` is used to suppress suppress if your eslint has the react hooks plugin.  I do.  If you checkout my branch and add all deps, you will see functionally these hooks will not work.